components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -218,6 +222,25 @@ class GeckoWebExtension(
         val geckoSession = (session as GeckoEngineSession).geckoSession
         return geckoSession.getWebExtensionActionDelegate(nativeExtension) != null
     }
+
+    /**
+     * See [WebExtension.getMetadata].
+     */
+    override fun getMetadata(): Metadata? {
+        return nativeExtension.metaData?.let {
+            Metadata(
+                name = it.name,
+                description = it.description,
+                developerName = it.creatorName,
+                developerUrl = it.creatorUrl,
+                homePageUrl = it.homepageUrl,
+                version = it.version,
+                permissions = it.permissions.toList(),
```

Comment:
We could use `Addon.permissionToTranslation` to localize this one.

https://github.com/mozilla-mobile/android-components/blob/e29cc1990413a93a1b243ae7f131bbe18a66a0e2/components/feature/addons/src/main/java/mozilla/components/feature/addons/Addon.kt#L120

Reply:
Yes, 👍 , but not here in the engine. That's for the follow-up when we integrate with `feature-addons`.

Reply:
👍